kratos/containers/pointer_vector_set.h

```diff
@@ -213,27 +210,18 @@ class PointerVectorSet final
      */
     pointer& operator()(const key_type& Key)
     {
-        ptr_iterator sorted_part_end;
-
-        if (mData.size() - mSortedPartSize >= mMaxBufferSize) {
-            Sort();
-            sorted_part_end = mData.end();
-        } else
-            sorted_part_end = mData.begin() + mSortedPartSize;
-
-        ptr_iterator i(std::lower_bound(mData.begin(), sorted_part_end, Key, CompareKey()));
-        if (i == sorted_part_end) {
-            mSortedPartSize++;
-            return *mData.insert(sorted_part_end, TPointerType(new TDataType(Key)));
+        auto itr_pos = std::lower_bound(mData.begin(), mData.end(), Key, CompareKey());
+        if (itr_pos == mData.end()) {
+            // insert a new element with id.
+            mData.push_back(TPointerType(new TDataType(Key)));
+            return mData.back();
+        } else if (EqualKeyTo(Key)(*itr_pos)) {
```

Comment:
same comment

Reply:
As explained above, with `std::lower_bound`, we always have to check for equality, because it returns an iterator to the first element which is greater than or equal to key.